### PR TITLE
More progress on moving body parts to SO

### PR DESCRIPTION
**Pull request by corp-0**
This time I started modifying the base class to make it more readable and support the new body parts.

I did a play test and I have no NRE, but no damage is done because we need to create the body part data and hook it into the Playerv3 prefabs. I'm waiting for you to attack the prefabs so we can agree on how to proceed.

- Finished implementation of the new body part class. All that is needed now is to actually create body part objects and assign them.
- Did a tame make over to the HealthSystem.cs class, we're using regions now to separate stuff and for easy recognition of, welp, regions in your code.
- Left a lot of TODO's for me so I can finish the logic and correct what we were doing before but can do better now.

Take a look and tell me what you think so we can progress on this.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Health/Living/HealthSystem.cs [hunk 1]
```diff
@@ -25,73 +24,62 @@ public abstract class HealthSystem : NetworkBehaviour, IHealth, IFireExposable,
...
-
-		/// <summary>
-		/// Server side, each mob has a different one and never it never changes
-		/// </summary>
-		public int mobID { get; private set; }
-
+		#region Inspector
 		[Tooltip("Max amount of HP this creature has overall.")]
-		public float MaxHealth = 100;
+		public float maxHealth = 100;
```
Comment (Cardioxo) [inline UnityProject/Assets/Scripts/Health/Living/HealthSystem.cs:29]: Why was this changed to camelCase, if it's intended to be a public field then it should be CamelCase

File: UnityProject/Assets/Scripts/Health/Living/HealthSystem.cs [hunk 2]
```diff
@@ -746,14 +783,91 @@ public void UpdateClientBodyPartStats(BodyPartType bodyPartType, float bruteDama
...
-				bodyPart.UpdateClientBodyPartStat(bruteDamage, burnDamage);
+				// bodyPart.UpdateClientBodyPartStat(bruteDamage, burnDamage);
 			}
 		}
+		#endregion
 
-		/// ---------------------------
-		/// MISC Functions:
-		/// ---------------------------
+		#region Event methods
```
Comment (Cardioxo) [inline UnityProject/Assets/Scripts/Health/Living/HealthSystem.cs:791]: I don't know if it's a good idea to put OnStartServer(), OnStartClient() and OnSpawnServer() in the bottom of the class. I can understand the want to group up methods, but it can be sort of confusing for a person looking in this class for the first time and not seeing all the Awake() / OnEnable() / OnStartServer() and so on right after all the variables, but instead having some of them in the bottom.

File: UnityProject/Assets/Scripts/Health/Living/HealthSystem.cs (hunk 1 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Health/Living/HealthSystem.cs:29]: Unity convention for serialized fields

File: UnityProject/Assets/Scripts/Health/Living/HealthSystem.cs (hunk 2 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Health/Living/HealthSystem.cs:791]: I can move them to the top. Maybe consider those as init methods or do their own category, “unity events”
